src/OSPSuite.Core/Domain/Services/FormulaTask.cs

```diff
@@ -322,31 +321,66 @@ public void ExpandDynamicFormulaIn(IContainer rootContainer)
                throw new CircularReferenceInSumFormulaException(dynamicFormula.Name, entityUsingFormula.Name);
 
             dynamicFormula.Criteria = updateDynamicFormulaCriteria(dynamicFormula, entityUsingFormula);
-            entityUsingFormula.Formula = dynamicFormula.ExpandUsing(allFormulaUsable, _objectPathFactory, _objectBaseFactory);
+
+            //this needs to be done after updating the criteria as dynamic tags might be added to the criteria and would 
+            //render the descriptor map list invalid
+            var allFormulaUsableDescriptor = allFormulaUsable.ToEntityDescriptorMapList();
+            entityUsingFormula.Formula = dynamicFormula.ExpandUsing(allFormulaUsableDescriptor, _objectPathFactory, _objectBaseFactory);
          });
       }
 
       private DescriptorCriteria updateDynamicFormulaCriteria(DynamicFormula formula, IUsingFormula usingFormula)
       {
-         //we need to replace IN PARENT criteria with actual criteria matching the parent of the usingFormula
          var criteria = formula.Criteria;
-         var allInParentTags = criteria.Where(x => x.IsAnImplementationOf<InParentCondition>()).ToList();
          var parent = usingFormula.ParentContainer;
-         if (!allInParentTags.Any() || parent == null)
+         if (parent == null)
+            return criteria;
+
+
+         var modifiedCriteria = modifyInParentFormulaCriteria(criteria, parent);
+         modifiedCriteria = modifyInChildrenFormulaCriteria(modifiedCriteria, parent);
+
+         return modifiedCriteria;
+      }
+
+      private DescriptorCriteria modifyInParentFormulaCriteria(DescriptorCriteria criteria, IContainer parent) => modifyDynamicCriteria<InParentCondition>(criteria, parent).criteria;
+
+      private DescriptorCriteria modifyInChildrenFormulaCriteria(DescriptorCriteria criteria, IContainer parent) {
+         var (modifyCriteria, modified) = modifyDynamicCriteria<InChildrenCondition>(criteria, parent);
+         if(!modified)
             return criteria;
 
+         //in case of IN CHILDREN, we need to exclude the direct children from the parent from the search.
+         //The only way to do this is to add a condition to exclude the children explicitly. We create a random string as tag
+         //to avoid collision and set it in all DIRECT only children  so that they will be excluded 
+
+         var uniqueTag = ShortGuid.NewGuid();
```

Comment:
tha'ts the gist of the logic. I create a tag, add it to all direct children and modify the criteria of the sum formula to exclude the tag explicitrely. I do not know how to do it otherwise